spec/Utils.spec.js

```
@@ -287,4 +288,159 @@ describe('Utils', () => {
+
+  describe('isPromise', () => {
+    it('should return true for a Promise', () => {
+      expect(Utils.isPromise(Promise.resolve())).toBe(true);
+    });
+    it('should return true for a cross-realm Promise', () => {
+      const crossRealmPromise = vm.runInNewContext('Promise.resolve()');
+      expect(crossRealmPromise instanceof Promise).toBe(false);
+      expect(Utils.isPromise(crossRealmPromise)).toBe(true);
+    });
+    it('should return true for a thenable', () => {
+      expect(Utils.isPromise({ then: () => {} })).toBe(true);
```

Biome (2.4.7)

[error] Do not add then to an object.

(lint/suspicious/noThenProperty)